### PR TITLE
Core data: Fix some typing issues

### PR DESCRIPTION
**Pull request by dmsnell**
Part of #39211

## Description
Fixes a few small issues with the JSDoc types in `@wordpress/core-data`.

 - `undefined` was the wrong return type for the given files. removing that type to remove the error and leaving them untyped.
 - acknowledge that the `undoIgnore` parameter is optional for `editEntityRecord`
 - return type of `__experimentalBatch` is a function that returns a `Promise`, not a `Promise`

## Testing Instructions
Audit the types. As a type-only change this won't impact the built files

## Types of changes
This is a type-only change and won't impact the build.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/core-data/src/actions.js [hunk 1]
```diff
@@ -607,8 +603,8 @@ export const saveEntityRecord = (
  *                         `saveEntityRecord`, `saveEditedEntityRecord`, and
  *                         `deleteEntityRecord`.
  *
- * @return {Promise} A promise that resolves to an array containing the return
- *                   values of each function given in `requests`.
+ * @return {(thinkArgs: Object) => Promise} A promise that resolves to an array containing the return
```
Comment (adamziel) [inline packages/core-data/src/actions.js]: Why `thinkArgs`?

**Review by adamziel — APPROVED** [review]
I have one nitpick, but looks good otherwise.

File: packages/core-data/src/actions.js (hunk 1 above)
Comment (adamziel) [reply packages/core-data/src/actions.js]: Looking at the original PR, I think this was meant to say `thunk` – I went ahead and updated it
```suggestion
 * @return {(thunkArgs: Object) => Promise} A promise that resolves to an array containing the return
```

Comment (dmsnell) [reply packages/core-data/src/actions.js]: Thanks for the update; you were spot on - thunkArgs - We could have made a better type here but mostly I want to get things set so we don’t have errors once we turn on TypeScript

**Pull request merged**